components/Discussion/Discussion.js

```diff
@@ -31,16 +32,23 @@ const Discussion = ({
       : discussionId === GENERAL_FEEDBACK_DISCUSSION_ID
       ? 'DATE'
       : 'AUTO')
+  const tag = query.tag
 
   const depth = board ? 1 : DEFAULT_DEPTH
 
   return (
     <div data-discussion-id={discussionId}>
       {!rootCommentOverlay && (
         <>
+          <TagFilter
+            discussionId={discussionId}
+            depth={depth}
+            orderBy={orderBy}
+          />
           <DiscussionCommentComposer
             discussionId={discussionId}
             orderBy={orderBy}
+            selectedTag={tag}
```

Comment:
Since `DiscussionCommentComposer` also uses `withDiscussionComments` we also need to pass `tag` with the name that the query expects. Also feel free to refactor the query for a unified name like `selectedTag` or `activeTag`.
```suggestion
            tag={tag}
```